src/reasoning/PrologClient.cpp

```diff
@@ -39,62 +42,208 @@ uint PrologClient::Instance::Get_id() const noexcept {
+}
+
+struct player{
+    std::string _name;
+    uint _num; //Nao is always 1, after that is 2,3,4,5,.....
+}
+
+
+struct round{
+    Instance _round;
+    turn _current_turn;
+    uint _timeInstance
+
+}
+
+void PrologClient::save_action_TurnCard(const TurnCard& action,uint timeInstant){
     //create prolog instance of TurnCard
+    Instance turn_card("TurnOneCard","Turn_Card");
+    create_instance(turn_card);
+    //assert Timestamp_instance as property from TurnCard
+    Instance timeStamp("TimeStamp","TimeStamp" + std::to_string(timeInstant));
+    create_instance(timeStamp);
+    assert_timeStamp_property(turn_card,timeStamp);
+    //assert CardPosition as an instance and linkt it to turn_card as two separate propertys
+    Instance card_position("CardPosition","Card_position" +std::to_string(timeInstant));
+    Property x_pos("hasXCoordinate",std::to_string(TurnCard._x));
+    Porperty y_pos("hasYCoordinate",std::to_string(TurnCard._y));
+    assert_property(card_position,x_pos);
+    assert_property(card_position,y_pos);
+    assert_property(turn_card,card_position);
+
+    //So now there should be a Instance od TurnCard which has as propertys: a timestamp(instance of timestamp) and a postion(instance of CardPosition)
+    
+
 }
 
-struct ActionTyp2 {
+
+void PrologClient::save_action_turn_equal_cards(const EqualCard& action,uint timeInstant){
+    save_action_TurnCard(action._Card1, timeInstant);
+    save_action_TurnCard(action._Card2, timeInstant);
+    Instance turn_equal_cards("TurnEqualCards","Turn_Equal_Cards");
+    //assert Timestamp_instance as property from turn_equal_cards
+    Instance timeStamp("TimeStamp","TimeStamp" + std::to_string(timeInstant));
+    create_instance(timeStamp);
+    assert_timeStamp_property(turn_equal_cards,timeStamp); 
 
 }
 
-void save_action(const ActionTyp2& action){
+
+void PrologClient::save_action_turn_two_unkown_cards(const unknownCard& Card1, consta unknownCard& Card2,uint TimeInstance){
+    
+    save_action_TurnCard(Card1._unknown_card, timeInstant);
+    save_action_TurnCard(Card2._unknown_card, timeInstant);
 
 }
 
+void PrologClient::instanciate_one_unknowncard(uint i,uint j,uint count){
+    //sdt::string name<<"unknown_card_"+std::to_string(i)+std:to_string(j);
+    Instance unknown_card_ins[count]("UnknownCard","UnkownCard_"+std::to_string(i)+std:to_string(j));
+    create_instance(unknown_card_ins[count]);
+    Instance card_position("CardPosition","Card_position"+std::to_string(i)+std:to_string(j));
+    Property x_pos("hasXCoordinate",std::to_string(i));
+    unknonw_card_ins[count]._position._x=i;
+    Porperty y_pos("hasYCoordinate",std::to_string(j));
+    unknonw_card_ins[count]._position._y=j;
+    assert_property(card_position,x_pos);
+    assert_property(card_position,y_pos);
+    assert_property(unkown_card_ins[count],card_position);
+}
+
+void PrologClient::instanciate_all_unknownCards(){
+    //Do at the beginnnig 
+    Instance unknown_card_ins[12];
```

Comment:
(1) arent these 9 cards?
(2) whats the use?

Reply:
already changed it to be 12
so that at the begining of the game we have 12 unknown cards and 0 known